include/aspect/parameters.h

```diff
@@ -702,7 +703,14 @@ namespace aspect
      * @}
      */
 
-
+    /**
+     * Adjoint problem
+     */
+    std::string        adjoint_input_file;
+    bool               read_in_points;
```

Comment:
I ended up removing these for this first pull request because including specific data locations will need some more code adjustment. I thought maybe it's better to keep this first pull request to a minimum and then extend it later with more options.